### DISCUSSION TITLE
How to modify data after query is completed?

### DISCUSSION THREAD

**Discussion by taneba** [urql-graphql/urql#1194] [Q&A] [answered]
I'm using default Document Cache behavior, and I want to transform data after retrieve data (e.g. string to Date object).
I know it can be easily done by using GraphCache, but I don't want to change cache strategy for that.

Now I'm doing useEffect, but is there any better solution?

**Comment by JoviDeCroock** [comment, marked as answer]
I think a better solution would be to utilize `useMemo`, that way you can react synchronously to this result rather than rendering once with your normal field and once with your interpretation of it.

Out of curiosity, what's stopping you from doing:

```
const [result] = useQuery({ query: `query { me { birthDate } }` });

return <p>{new Date(birthDate)}</p>
```

This can save you the complexity of having to duplicate said state.

**Reply by taneba** [reply]
Thanks for quick response @JoviDeCroock 
yeah useMemo should be better, thanks.

The reason I want to do it close to data fetch layer is because our backend api format is like `birthDate: "2020/12/09"`, which can not pass new Date() directly, and also because we are doing date calculation using date-fns in multiple layer.

Anyways, in my case `useMemo` seems to be the better solution.
Thanks, really appreciated for your help!

**Comment by shimahi** [comment]
Hi.

I was also looking for a way to format the fetched data while using Document Cache in this way.

I am very curious about how to useMemo, can you provide an example?
Is it set in the context option of the useQuery execution code?

**Reply by taneba** [reply]
Here's an example using Jovi's code.
Actually it's not an urql specific technique, but using useMemo like this way can reduce unnecessary code.

```
const [result] = useQuery({ query: `query { me { birthDate } }` });

const modifiedData = useMemo(() => ({
  ...res.data,
  birthDate: new Date(birthDate)
}),[result.data])
```

**Reply by shimahi** [reply]
Oh! It's very useful.
Really thank you!
